frontend/app/psi.ts

```diff
@@ -0,0 +1,96 @@
+import { base64 } from "rfc4648";
+
+const sodium = require("libsodium-wrappers-sumo");
+
+type ServerResponse = {
+    userPassword: string;
+    breachedPasswords: string[];
+};
+
+export function hashToPoint(input: string): Uint8Array {
+    const hash = sodium.crypto_generichash(
+        sodium.crypto_core_ristretto255_HASHBYTES,
+        sodium.from_string(input)
+    );
+    return sodium.crypto_core_ristretto255_from_hash(hash);
+}
+
+/**
+ *
+ * @param input the string to be encrypted
+ * @returns input with a secret key applied and the key's inverse
+ */
+export function applySeed(input: string): [Uint8Array, Uint8Array] {
+    // generate random seed
+    const seed = sodium.crypto_core_ristretto255_scalar_random();
+    // get seed inverse
+    const seedInverse =
+        sodium.crypto_core_ristretto255_scalar_invert(seed);
+    const point = hashToPoint(input);
+    // apply seed
+    const seededPassword = sodium.crypto_scalarmult_ristretto255(
+        seed,
+        point
+    );
+    console.log("seeded password: ", new TextDecoder().decode(seededPassword.buffer));
+    return [seededPassword, seedInverse];
+}
+
+function computeIntersection(
+    data: ServerResponse,
+    aInverse: Uint8Array
+): boolean {
+    const password = data.userPassword;
```

Comment:
**nitpick:** could just decode base64 here to be consistent
`const password = base64.parse(data.userPassword)`